crash importing lxc

Similar to #1331.

I'm getting a crash importing a `proxmox_lxc` from Proxmox 7.4-17.

`terraform version`:

```
Terraform v1.12.1
on darwin_arm64
+ provider registry.terraform.io/telmate/proxmox v3.0.2-rc01
```

`main.tf`:

```terraform
terraform {
  required_providers {
    proxmox = {
      source  = "Telmate/proxmox"
      version = "3.0.2-rc01"
    }
  }
}

provider "proxmox" {
  pm_log_enable = true
  pm_log_file   = "terraform-plugin-proxmox.log"
  pm_debug      = true
  pm_log_levels = {
    _default    = "debug"
    _capturelog = ""
  }

  …
}

resource "proxmox_lxc" "nas-mk2" {
}
```

output:

```
$ terraform import proxmox_lxc.nas-mk2 quongo/lxc/106
proxmox_lxc.nas-mk2: Importing from ID "quongo/lxc/106"...
proxmox_lxc.nas-mk2: Import prepared!
  Prepared proxmox_lxc for import
proxmox_lxc.nas-mk2: Refreshing state... [id=quongo/lxc/106]
╷
│ Error: Request cancelled
│
│ The plugin.(*GRPCProvider).ReadResource request was cancelled.
╵


Stack trace from the terraform-provider-proxmox_v3.0.2-rc01 plugin:

panic: interface conversion: interface {} is nil, not *schema.Set

goroutine 89 [running]:
github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/node.Terraform(0x140005fa510, {0x1400080a8b0, 0x6})
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/node/terraform.go:15 +0x240
github.com/Telmate/terraform-provider-proxmox/v2/proxmox._resourceLxcRead({0x1016af8f8, 0x140000ef420}, 0x140005fa510, {0x1014f7da0?, 0x1400060c3c0})
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/resource_lxc.go:813 +0x250
github.com/Telmate/terraform-provider-proxmox/v2/proxmox.resourceLxcRead({0x1016af8f8?, 0x140000ef420?}, 0x1176592e000?, {0x1014f7da0?, 0x1400060c3c0?})
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/resource_lxc.go:790 +0x2c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x140003d0600, {0x1016af850, 0x14000170a80}, 0x140005fa510, {0x1014f7da0, 0x1400060c3c0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.37.0/helper/schema/resource.go:866 +0xe4
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0x140003d0600, {0x1016af850, 0x14000170a80}, 0x140004f9ba0, {0x1014f7da0, 0x1400060c3c0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.37.0/helper/schema/resource.go:1162 +0x400
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0x1400058e030, {0x1016af850?, 0x140001709c0?}, 0x1400060d9a0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.37.0/helper/schema/grpc_provider.go:908 +0x9bc
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0x1400027a000, {0x1016af850?, 0x14000362270?}, 0x140000eeb60)
        github.com/hashicorp/terraform-plugin-go@v0.28.0/tfprotov5/tf5server/server.go:859 +0x210
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x101681520, 0x1400027a000}, {0x1016af850, 0x14000362270}, 0x140000e5200, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.28.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:651 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x1400025e400, {0x1016af850, 0x140003621e0}, 0x1400010f2c0, 0x14000517020, 0x101ceb198, 0x0)
        google.golang.org/grpc@v1.72.1/server.go:1405 +0xc9c
google.golang.org/grpc.(*Server).handleStream(0x1400025e400, {0x1016b0150, 0x140000b0000}, 0x1400010f2c0)
        google.golang.org/grpc@v1.72.1/server.go:1815 +0x900
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.72.1/server.go:1035 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 39
        google.golang.org/grpc@v1.72.1/server.go:1046 +0x138

Error: The terraform-provider-proxmox_v3.0.2-rc01 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

Slightly sanitized log: 

[terraform-plugin-proxmox.log](https://github.com/user-attachments/files/20725204/terraform-plugin-proxmox.log)

This `500` stands out calling `GET /api2/json/cluster/ha/resources/106`:

```
2025/06/13 06:45:16 >>>>>>>>>> REQUEST:
GET /api2/json/cluster/ha/resources/106 HTTP/1.1
Host: …
User-Agent: Go-http-client/1.1
Accept: application/json
Authorization: …
Accept-Encoding: gzip

2025/06/13 06:45:16 <<<<<<<<<< RESULT:
HTTP/1.1 500 no such resource 'ct:106'
Connection: close
Content-Length: 13
Cache-Control: max-age=0
Content-Type: application/json;charset=UTF-8
Date: Fri, 13 Jun 2025 10:45:16 GMT
Expires: Fri, 13 Jun 2025 10:45:16 GMT
Pragma: no-cache
Server: pve-api-daemon/3.0

{"data":null}
```

Hello, registering what happened here...
Using Proxmox 8.4.1

Import with version 3.0.1-rc6 was successfull...

Tried to import in version3.0.2-rc01gave the following error:

$terraform import proxmox_lxc.ZabbixServer-3 "homeserver-n3/lxc/117"

terraform import proxmox_lxc.ZabbixServer-3 "homeserver-n3/lxc/117"
proxmox_lxc.ZabbixServer-3: Importing from ID "homeserver-n3/lxc/117"...
proxmox_lxc.ZabbixServer-3: Import prepared!
  Prepared proxmox_lxc for import
proxmox_lxc.ZabbixServer-3: Refreshing state... [id=homeserver-n3/lxc/117]
╷
│ Error: Request cancelled
│ 
│ The plugin.(*GRPCProvider).ReadResource request was cancelled.
╵


Stack trace from the terraform-provider-proxmox_v3.0.2-rc01 plugin:

panic: interface conversion: interface {} is nil, not *schema.Set

goroutine 97 [running]:
github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/node.Terraform(0xc0007055f0, {0xc0007746f0, 0xd})
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/node/terraform.go:15 +0x234
github.com/Telmate/terraform-provider-proxmox/v2/proxmox._resourceLxcRead({0x1034bb8, 0xc0005930a0}, 0xc0007055f0, {0xd287e0?, 0xc000259180})
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/resource_lxc.go:813 +0x33c
github.com/Telmate/terraform-provider-proxmox/v2/proxmox.resourceLxcRead({0x1034bb8?, 0xc0005930a0?}, 0x1176592e000?, {0xd287e0?, 0xc000259180?})
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/resource_lxc.go:790 +0x2b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc0004c1d00, {0x1034b10, 0xc00063db60}, 0xc0007055f0, {0xd287e0, 0xc000259180})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.37.0/helper/schema/resource.go:866 +0x119
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0004c1d00, {0x1034b10, 0xc00063db60}, 0xc0000b4750, {0xd287e0, 0xc000259180})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.37.0/helper/schema/resource.go:1162 +0x52a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc000376000, {0x1034b10?, 0xc00063daa0?}, 0xc0006724b0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.37.0/helper/schema/grpc_provider.go:908 +0xb6d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc0002840a0, {0x1034b10?, 0xc00063cfc0?}, 0xc000592af0)
        github.com/hashicorp/terraform-plugin-go@v0.28.0/tfprotov5/tf5server/server.go:859 +0x2f3
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0xeae720, 0xc0002840a0}, {0x1034b10, 0xc00063cfc0}, 0xc0001a1980, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.28.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:651 +0x1a9
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000266400, {0x1034b10, 0xc00063cf30}, 0xc00009b080, 0xc00059fd40, 0x16a2f18, 0x0)
        google.golang.org/grpc@v1.72.1/server.go:1405 +0x1036
google.golang.org/grpc.(*Server).handleStream(0xc000266400, {0x10354d0, 0xc0005fa000}, 0xc00009b080)
        google.golang.org/grpc@v1.72.1/server.go:1815 +0xb88
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.72.1/server.go:1035 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 38
        google.golang.org/grpc@v1.72.1/server.go:1046 +0x11d

Error: The terraform-provider-proxmox_v3.0.2-rc01 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

--- a/proxmox/Internal/resource/guest/node/schema.go
+++ b/proxmox/Internal/resource/guest/node/schema.go
@@ -1,3 +1,4 @@
+// Package node provides functions to handle PVE nodes for Guest systems in Terraform.
 package node
 
 import (

--- a/proxmox/Internal/resource/guest/node/sdk.go
+++ b/proxmox/Internal/resource/guest/node/sdk.go
@@ -9,7 +9,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-const error_NoNodeConfigured = "no target node specified"
+const errorNoNodeConfigured = "no target node specified"
 
 func SdkUpdate(d *schema.ResourceData, current pveAPI.NodeName) (pveAPI.NodeName, error) {
 	if node, ok := d.GetOk(RootNode); ok {
@@ -19,7 +19,7 @@ func SdkUpdate(d *schema.ResourceData, current pveAPI.NodeName) (pveAPI.NodeName
 	currentNode := string(current)
 	switch len(nodes) {
 	case 0:
-		return "", errors.New(error_NoNodeConfigured)
+		return "", errors.New(errorNoNodeConfigured)
 	case 1:
 		if currentNode != nodes[0].(string) {
 			return pveAPI.NodeName(nodes[0].(string)), nil
@@ -40,7 +40,7 @@ func SdkCreate(d *schema.ResourceData) (pveAPI.NodeName, error) {
 	nodes := d.Get(RootNodes).(*schema.Set).List()
 	switch len(nodes) {
 	case 0:
-		return "", errors.New(error_NoNodeConfigured)
+		return "", errors.New(errorNoNodeConfigured)
 	case 1:
 		return pveAPI.NodeName(nodes[0].(string)), nil
 	}

--- a/proxmox/Internal/resource/guest/node/terraform.go
+++ b/proxmox/Internal/resource/guest/node/terraform.go
@@ -12,10 +12,12 @@ func Terraform(currentNode pveAPI.NodeName, d *schema.ResourceData) {
 		d.Set(RootNode, current)
 		return
 	}
-	nodes := d.Get(RootNodes).(*schema.Set).List()
-	if inArray(nodes, current) {
-		d.Set(RootNodes, nodes)
-		return
+	if v, ok := d.GetOk(RootNodes); ok && v != nil {
+		nodes := v.(*schema.Set).List()
+		if inArray(nodes, current) {
+			d.Set(RootNodes, nodes)
+			return
+		}
 	}
-	d.Set(RootNodes, []interface{}{current})
+	d.Set(RootNodes, []any{current})
 }

